### PR TITLE
Bug 1847141 - Restrict download broadcasts to app process.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1847141

### CODE REVIEW

**Comment by mcarare** [comment]
@jonalmeida does r-b have the same download flow as Fenix/ Focus?

**Comment by jonalmeida** [comment]
Yes, it should, if my memory still holds. 🤔

File: android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt [hunk 1]
```diff
@@ -840,7 +840,7 @@ abstract class AbstractFetchDownloadService : Service() {
         intent.putExtra(EXTRA_DOWNLOAD_STATUS, getDownloadJobStatus(downloadState))
         intent.putExtra(EXTRA_DOWNLOAD_ID, downloadState.state.id)
 
-        context.sendBroadcast(intent)
+        context.sendBroadcast(intent, "org.mozilla.permission.RECEIVE_BROADCAST")
```
Comment (jonalmeida) [inline android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt:843]: Consider making this a public val or something that lets you add a kdoc that this is needed to be added to an application's manifest.

**Review by jonalmeida — APPROVED** [review]
Nice fix!

File: android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt (hunk 1 above)
Comment (jonalmeida) [reply android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt]: @mcarare your latest patch looks good to me. Thanks!

**Pull request merged**